alchemy/test/aws/function.test.ts

```diff
@@ -514,5 +514,158 @@ describe("AWS Resources", () => {
+          runtime: "nodejs20.x",
+          tags: {
+            Environment: "test",
+          },
+          url: {
+            authType: "NONE",
+            // Default invokeMode is BUFFERED if not specified
+            cors: {
+              allowOrigins: ["*"],
+              allowMethods: ["GET", "POST"],
+              allowHeaders: ["Content-Type"],
+            },
+          },
+        });
+
+        // Verify function was created with URL
+        expect(func.arn).toBeTruthy();
+        expect(func.state).toBe("Active");
+        expect(func.functionUrl).toBeTruthy();
+
+        // Test function URL invocation (default BUFFERED mode)
+        const testEvent = { test: "buffered-mode" };
+        const response = await fetch(func.functionUrl!, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(testEvent),
+        });
+
+        expect(response.status).toBe(200);
+        const responseBody = await response.json();
+        expect(responseBody.message).toBe("Hello from bundled handler!");
+        expect(responseBody.event).toEqual(testEvent);
+
+        // Update function to explicitly set BUFFERED mode
+        func = await Function(functionName, {
+          functionName,
+          bundle,
+          roleArn: role.arn,
+          handler: "handler.handler",
+          runtime: "nodejs20.x",
+          tags: {
+            Environment: "test",
+          },
+          url: {
+            authType: "NONE",
+            invokeMode: "BUFFERED", // Explicitly set BUFFERED
+            cors: {
+              allowOrigins: ["*"],
+              allowMethods: ["GET", "POST"],
+              allowHeaders: ["Content-Type"],
+            },
+          },
+        });
+
+        // Verify function still has URL
+        expect(func.functionUrl).toBeTruthy();
+
+        // Now update to RESPONSE_STREAM mode
+        func = await Function(functionName, {
+          functionName,
+          bundle,
+          roleArn: role.arn,
+          handler: "handler.handler",
+          runtime: "nodejs20.x",
+          tags: {
+            Environment: "test",
+          },
+          url: {
+            authType: "NONE",
+            invokeMode: "RESPONSE_STREAM", // Change to streaming mode
+            cors: {
+              allowOrigins: ["*"],
+              allowMethods: ["GET", "POST"],
+              allowHeaders: ["Content-Type"],
+            },
+          },
+        });
+
+        // Verify function still has URL
+        expect(func.functionUrl).toBeTruthy();
+
+        // Test function URL invocation (now in RESPONSE_STREAM mode)
+        const streamTestEvent = { test: "response-stream-mode" };
+        const streamResponse = await fetch(func.functionUrl!, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(streamTestEvent),
+        });
```

Comment:
should we be testing that it does actually stream? Is that possible?

Reply:
i think we could,
perhaps something along those lines? 
```
// Check transfer encoding
const isChunked = response.headers.get('transfer-encoding') === 'chunked';

// Check for readable stream body
const isStream = response.body instanceof ReadableStream;

// Consuming the tream 
const reader = response.body.getReader();
...
  ```

Reply:
Ah yeah, the header is worth testing. Not sure consuming the stream is helpful since won't even a standard JSON response be streamable?